testsupport/space.go

```diff
@@ -0,0 +1,95 @@
+package testsupport
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-e2e/testsupport/tiers"
+	"github.com/codeready-toolchain/toolchain-e2e/testsupport/wait"
+	"github.com/gofrs/uuid"
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/stretchr/testify/require"
+)
+
+// NewSpace initializes a new Space object with the given value. If the targetCluster is nil, then the Space.Spec.TargetCluster is not set.
+func NewSpace(namespace, name, tierName string, targetCluster *wait.MemberAwaitility) *toolchainv1alpha1.Space {
+	space := &toolchainv1alpha1.Space{
+		ObjectMeta: v1.ObjectMeta{
+			Namespace: namespace,
+			Name:      name,
+		},
+		Spec: toolchainv1alpha1.SpaceSpec{
+			TierName: tierName,
+		},
+	}
+	if targetCluster != nil {
+		space.Spec.TargetCluster = targetCluster.ClusterName
+	}
+	return space
+}
+
+// GenerateName appends generated UUID to the given string
+func GenerateName(prefix string) string {
```

Comment:
Should this func be moved to a more common location since it's not specific to spaces?

Reply:
I wasn't sure where to put it, so I left it there. But you are right - moved 5d18035